godeltaprof/README.md

```diff
@@ -1 +1,98 @@
 # godeltaprof
+
+godeltaprof is an efficient delta profiler for memory, mutex, block.
+
+# Why
+
+In golang, allocation, mutex and block profiles are cumulative - they only grow over time and show allocations happened since the beginning of the running program.
+Not only values grow, but the size of the profile itself grows as well. It could grow up to megabytes in size for long-running processes. These megabytes profiles are called huge profiles in this document.
+
+In many cases it is much more useful to see the difference between two points in time.
+
+There is a way to do it with the original runtime/pprof package, they call it delta profile, it requires passing seconds argument to pprof endpoint query.
+
+```
+go tool pprof http://localhost:6060/debug/pprof/heap?seconds=30
+```
+
+What this does:
+1. dump profile p0
+2. sleep
+3. dump profile p1
+4. decompress and parse protobuf p0
+5. decompress and parse protobuf p1
+6. subtract p0 from p1
+7. serialize protobuf and compress the result
+
+The resulting profile is *usually* much smaller(p0 may be megabytes, while result is usually tens of kilobytes).
+
+There are number of issues with this approach.
+1. Heap profile contains both allocation values and inuse values. Inuse values are not cumulative. So inuse values are corrupted by the subtraction.
+Note: it can be fixed if runtime/pprof package would do the following `p0.ScaleN([]float64{-1,-1,0,0})` instead of `p0.Scale(-1)` - that would substract allocation values and zero out inuse values in p0.
+2. It requires dumping two profiles.
+3. It produces a lot of allocations putting pressure on GC.
```

Comment:
Is "inuse value" normally one word when referring to Golang? In normal grammar, in-use would be hyphenated because it's a compound adjective (in-use (descriptive - adjective) describes values (noun)). 
```suggestion
The resulting profile is *usually* much smaller (`p0` may be megabytes, while result is usually tens of kilobytes).

There are number of issues with this approach:

1. Heap profile contains both allocation values and in-use values. In-use values are not cumulative. In-use values are corrupted by the subtraction.
  **Note:** It can be fixed if runtime/pprof package uses `p0.ScaleN([]float64{-1,-1,0,0})`, instead of `p0.Scale(-1)` - that would subtract allocation values and zero out in-use values in `p0`.
2. It requires dumping two profiles.
3. It produces a lot of allocations putting pressure on GC.
```

Reply:
@knylander-grafana when I referred to "inuse" I referred to "inuse_space" and "inuse_objects" profling types from Golang pprof 

https://github.com/golang/go/blob/master/src/runtime/pprof/pprof.go#L127